client/my-sites/ads/main.jsx

```diff
@@ -223,18 +222,30 @@ const AdsMain = React.createClass( {
 } );
 
 export default connect(
-	( state, ownProps ) => ( {
-		requestingWordAdsApproval: isRequestingWordAdsApprovalForSite( state, ownProps.site ),
-		wordAdsError: getWordAdsErrorForSite( state, ownProps.site ),
-		wordAdsSuccess: getWordAdsSuccessForSite( state, ownProps.site ),
-		isUnsafe: isSiteWordadsUnsafe( state, ownProps.site.ID ),
-		isRequestingWordadsStatus: isRequestingWordadsStatus( state, ownProps.site.ID )
-	} ),
+	( state ) => {
+		const site = getSelectedSite( state );
+		const siteId = getSelectedSiteId( state );
+
+		return {
+			site,
+			siteId,
+			siteSlug: getSelectedSiteSlug( state ),
+			requestingWordAdsApproval: isRequestingWordAdsApprovalForSite( state, site ),
+			wordAdsError: getWordAdsErrorForSite( state, site ),
+			wordAdsSuccess: getWordAdsSuccessForSite( state, site ),
+			isUnsafe: isSiteWordadsUnsafe( state, siteId ),
+			isRequestingWordadsStatus: isRequestingWordadsStatus( state, siteId )
+		};
+	},
 	{ requestWordAdsApproval, dismissWordAdsError },
 	( stateProps, dispatchProps, parentProps ) => Object.assign(
 		{},
 		dispatchProps,
-		{ requestWordAdsApproval: () => ( ! stateProps.requestingWordAdsApproval ) ? dispatchProps.requestWordAdsApproval( parentProps.site.ID ) : null },
+		{
+			requestWordAdsApproval: () => ( ! stateProps.requestingWordAdsApproval )
+				? dispatchProps.requestWordAdsApproval( stateProps.siteId )
+				: null
+		},
```

Comment:
Good call, addressed in https://github.com/Automattic/wp-calypso/pull/13119/commits/7073493086980efd0d2ec4a3f839adff9c9d3060.